node-wasm/src/ports.rs

```diff
@@ -211,13 +211,17 @@ impl WorkerClient {
             .post_message(&command_value)
             .context("could not post message")?;
 
-        let worker_response = response_channel
-            .recv()
-            .await
-            .expect("response channel should never drop")
-            .context("error executing command")?;
+        loop {
+            let worker_response = response_channel
+                .recv()
+                .await
+                .expect("response channel should never drop")
+                .context("error executing command")?;
 
-        Ok(worker_response)
+            if !worker_response.is_internal_pong() {
+                return Ok(worker_response);
+            }
```

Comment:
I think this is more correct. In this case `InternalPong` will be returned if ping was requested.

```suggestion
            // Skip InternalPong if requested command was not InternalPing.
            // We use this because ping is meant to be used with timeout but the server might
            // reply with pong after the timeout is reached.
            if worker_response.is_internal_pong() && !matches!(&command, NodeCommand::InternalPing) {
                continue;
            }
            
            return Ok(worker_response);
```

Reply:
nice catch